pymeasure/instruments/temptronic/temptronic_base.py

```diff
@@ -0,0 +1,826 @@
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+"""
+Implementation of an interface class for ThermoStream® Systems devices.
+
+Reference Document for implementation:
+
+ATS-515/615, ATS 525/625 & ATS 535/635 ThermoStream® Systems
+Interface & Applications Manual
+Revision E
+September, 2019
+
+Here the list of NOT implemented cmds:
+
+Command      | imp | Description
+#############+#####+#################################################################
+*TST?        | no  | Self test (dummy, always returns 0, meaning “passed”).
+-------------+-----+-----------------------------------------------------------------
+*OPC         | no  | Not implemented
+*OPC?        |     |
+-------------+-----+-----------------------------------------------------------------
+*WAI         | no  | Not implemented
+-------------+-----+-----------------------------------------------------------------
+!            | no  | Device clear – clears the serial communications subsystem. The
+             |     | System echos back the “!” when the command has completed. If the
+             |     | “!” response is not received, the command should be retried.
+             |     | NOTE: This command is sent as a single character (no line feed
+             |     | terminator), and should never otherwise appear in a string sent to
+             |     | the System.
+-------------+-----+-----------------------------------------------------------------
+^            | no  | The System sends a “^” as a service request (SRQ) indicator in
+             |     | serial mode.
+             |     | NOTE: The “^” character never otherwise displays in a response
+             |     | string, and is sent as a single character.
+-------------+-----+-----------------------------------------------------------------
+"""
+
+from pymeasure.instruments.instrument import Instrument, log
+from pymeasure.instruments.validators import (
+        strict_discrete_set,
+        truncated_range,
+        strict_range
+        )
+from enum import Enum
+import time
+
+
+class StandardEventStatusCode(Enum):
```

Comment:
All those look like they should be `Enum.IntFlag` children, which would yield some nice benefits (see my remarks below)

Reply:
Switched to IntFlag and also introduced those get_processes for most relevant properties.